src/domain/transaction/evm/evm_transaction.rs

```diff
@@ -410,6 +444,41 @@ where
 
         let final_evm_data = updated_evm_data.with_signed_transaction_data(sig_result.into_evm()?);
 
+        // Validate the relayer has sufficient balance
```

Comment:
In case of insufficient balance in resubmit handler let's log this state and exit from handler by not resubmitting it at this point in time.

We do not want to mark tx as failed given that originally sent tx could be mined after some time or after gas prices goes down.

Also once prices go down resubmit could work.